### PR TITLE
Push down the limit operator to segcore

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #25561

### CODE REVIEW

File: internal/core/src/segcore/SegmentInterface.cpp [hunk 1]
```diff
@@ -178,4 +178,35 @@ SegmentInternalInterface::get_real_count() const {
...
+            bitset[offset] = false;
+        }
+    }
+}
+
+void
+SegmentInternalInterface::search_ids_filter(BitsetType& bitset,
+                                            const std::vector<int64_t>& offsets,
+                                            Timestamp timestamp) const {
+    BitsetType bitset_copy = bitset;
```
Comment (czs007) [inline internal/core/src/segcore/SegmentInterface.cpp:201]: remove this copy

File: internal/core/src/segcore/SegmentInterface.cpp [hunk 2]
```diff
@@ -178,4 +178,35 @@ SegmentInternalInterface::get_real_count() const {
     return res->fields_data()[0].scalars().long_data().data(0);
 }
 
+void
+SegmentInternalInterface::search_ids_filter(BitsetType& bitset,
+                                            Timestamp timestamp) const {
+    auto& timestamps = get_timestamps();
```
Comment (czs007) [inline internal/core/src/segcore/SegmentInterface.cpp:184]: how about add one quick test:
 if timestamps[bitset.size()] <= timestmap {
    return;
 }

**Comment by longjiquan** [comment]
I'll fix the comments in the next pr.

**Pull request merged**